src/topology/category/Profinite.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2020 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+-/
+
+import topology.category.CompHaus
+
+/-!
+# The category of Profinite Types
+
+We construct the category of profinite topological spaces,
+often called profinite sets -- perhaps they could be called
+profinite types in Lean.
+
+The type of profinite topological spaces is called `Profinite`. It has a category
+instance and is a fully faithful subcategory of `Top`. The fully faithful functor
+is called `Profinite_to_Top`.
+
+## Implementation notes
+
+A profinite type is defined to be a topological space which is
+compact, Hausdorff and totally disconnected.
+
+## TODO
+
+0. Link to category of projective limits of finite discrete sets.
+1. existence of products, limits(?), finite coproducts
+2. `Profinite_to_Top` creates limits?
+3. Clausen/Scholze topology on the category `Profinite`.
+
+## Tags
+
+profinite
+
+-/
+
+open category_theory
+
+/-- The type of profinite topological spaces. -/
+structure Profinite :=
+(to_Top : Top)
+[is_compact : compact_space to_Top]
+[is_t2 : t2_space to_Top]
+[is_td : totally_disconnected_space to_Top]
```

Comment:
```suggestion
[is_totally_disconnected : totally_disconnected_space to_Top]
```
A bit longer, but better readability when used outside the context of this file.